### PR TITLE
[six] initialize pyyaml: cache relevant pyobjects, and use C-extensions

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?

This PR does two things primarly:
1) Ensure we use the C-extensions for the pyyaml dumper and loader, `CSafeDumper` and `CSafeLoader`, if available. Using `safe_load` directly had the side-effect of bringing in the extra 30MB RSS product of the regex in the non-C loader/dumper. 
2) Cache the constantly used PyObjects to the `load`, `dump`, `loader` and `dumper` `pyyaml` objects, it makes no sense to grab them over and over again.

### Motivation

RSS liposuction 📉 

### Additional Notes

Since this should be called no matter what, it appeared most sensible to include the initialization in the `Two`/`Three` init routines. Please let me know if we don't agree on this. I believe exposing it with the API is pretty useless given that we need to pass no arguments from CGO, and this should always be called on `init()`.

The alternative would be to patch pyyaml to get rid of the regex altogether as proposed here: https://github.com/yaml/pyyaml/pull/124 - the `pyyaml` maintainers are still not sure about UCS4 implications or performance, so I felt this approach gave us the most control and better understanding over what was going on. 

Verified effective with a local install, not currently deployed anywhere, but feel free to validate with the gitlab artifacts :)

### CODE REVIEW

File: six/two/two.cpp [hunk 1]
```diff
@@ -168,6 +172,7 @@ py_info_t *Two::getPyInfo()
 
 bool Two::runSimpleString(const char *code) const
 {
+    PyErr_Clear();  // REMOVE
```
Comment (albertvaka) [inline six/two/two.cpp]: Remove

File: six/common/stringutils.c [hunk 2]
```diff
@@ -48,62 +56,103 @@ char *as_string(PyObject *object)
...
+    if (yload == NULL) {
         goto done;
     }
 
-    retval = PyObject_CallFunction(safe_load, "s", data);
+    // We try to use the C-extensions, if they're available, but it's a best effort
+    char c_loader_name[] = "CSafeLoader";
+    loader = PyObject_GetAttrString(yaml, c_loader_name);
+    if (loader == NULL) {
+        PyErr_Clear();
```
Comment (albertvaka) [inline six/common/stringutils.c]: Should we print something here? Like: "C loader not found, serialization will be slow"

**Review by albertvaka — APPROVED** [review]
LGTM, but what about other places that might be using load/dump/safe_load/safe_dump?

**Review by albertvaka — DISMISSED** [review]
LGTM. Are calls to load/dump/safe_load/safe_dump from checks already using those loaders? Is there any other place we might be using the python loader?

**Comment by truthbk** [comment]
> LGTM. Are calls to load/dump/safe_load/safe_dump from checks already using those loaders? Is there any other place we might be using the python loader?

@albertvaka we do ensure that all our checks + custom checks, unless explicitly disabled, use the safe loading/dumping facilities with a monkey patch on `pyyaml`. On top of that, if available (which would be the case for anything that is not a miscellaneous custom build) we will always default to the C-extensions for added performance. So I believe we're well covered in this sense.

File: six/common/stringutils.c (hunk 2 above)
Comment (arbll) [reply six/common/stringutils.c]: Agreed, but I don't think we have any logging mechanism in six...

**Review by arbll — APPROVED** [review]
LGTM

File: six/common/stringutils.c (hunk 2 above)
Comment (truthbk) [reply six/common/stringutils.c]: There's nothing straight-forward currently. We could log to stderr, but we'd just get something on the init journal/syslog, but nothing on the actual logs.

**Review by olivielpeau — COMMENTED** [review]
also reviewed the code, LGTM! 👍

**Pull request merged**